### PR TITLE
feat-new-image-features

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

**Comment by eldadfux** [comment]
@lohanidamodar I think the tests are failing on the storage endpoints.

**Comment by lohanidamodar** [comment]
@eldadfux the Tests are passing in Branch but failing in PR not sure. Do you need to re-run? Let me know

**Comment by eldadfux** [comment]
@lohanidamodar I restarted the tests. Please note, we need e2e tests here to validate the API endpoint behavior.

**Comment by lohanidamodar** [comment]
@eldadfux please restart the tests

File: app/controllers/api/storage.php [hunk 1]
```diff
@@ -242,13 +242,18 @@
     ->param('width', 0, new Range(0, 4000), 'Resize preview image width, Pass an integer between 0 to 4000.', true)
     ->param('height', 0, new Range(0, 4000), 'Resize preview image height, Pass an integer between 0 to 4000.', true)
     ->param('quality', 100, new Range(0, 100), 'Preview image quality. Pass an integer between 0 to 100. Defaults to 100.', true)
+    ->param('borderWidth', 0, new Range(0, 100), 'Preview image border. Pass an integer between 0 to 100. Defaults to 0.', true)
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:245]:
```suggestion
    ->param('borderWidth', 0, new Range(0, 100), 'Preview image border in pixels. Pass an integer between 0 to 100. Defaults to 0.', true)
```
Might be more clear

File: app/controllers/api/storage.php [hunk 2]
```diff
@@ -242,13 +242,18 @@
...
+    ->param('borderColor', '', new HexColor(), 'Preview image border color. Use a valid HEX color, no # is needed for prefix.', true)
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:246]:
```suggestion
    ->param('borderColor', '', new HexColor(), 'Preview image border color. Use a valid HEX color, # prefix is optional', true)
```

File: app/controllers/api/storage.php [hunk 3]
```diff
@@ -242,13 +242,18 @@
...
+    ->param('borderRadius', 0, new Range(0, 4000), 'Preview image border radius. Pass an integer between 0 to 4000.', true)
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:247]:
```suggestion
    ->param('borderRadius', 0, new Range(0, 4000), 'Preview image border radius in pixels. Pass an integer between 0 to 4000.', true)
```

File: app/controllers/api/storage.php [hunk 4]
```diff
@@ -242,13 +242,18 @@
...
+    ->param('opacity', 1, new Range(0,1), 'Preview image opacity. Only works with transparent images (png). Pass an integer between 0 to 1.', true)
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:248]:
```suggestion
    ->param('opacity', 1, new Range(0,1), 'Preview image opacity. Only works with transparent images (png). Pass a number between 0 to 1.', true)
```

File: app/controllers/api/storage.php [hunk 5]
```diff
@@ -242,13 +242,18 @@
...
+    ->param('rotation', 0, new Range(0,360), 'Preview image rotation. Pass an integer between 0 to 360.', true)
```
Comment (christyjacob4) [inline app/controllers/api/storage.php:249]:
```suggestion
    ->param('rotation', 0, new Range(0,360), 'Preview image rotation in degrees. Pass an integer between 0 and 360.', true)
```

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Looks Good @lohanidamodar  . Some minor suggestions

**Comment by eldadfux** [comment]
@lohanidamodar when @christyjacob4 are addressed we can merge this PR to 0.8

**Comment by lohanidamodar** [comment]
@eldadfux This should be ready!

**Comment by lohanidamodar** [comment]
@eldadfux pushed small fix for border with border radius issue. Need the newer version of utopia-php/image released with the latest PR merged.

**Comment by eldadfux** [comment]
@lohanidamodar we need to make sure the PR is using https://github.com/utopia-php/image v0.2.1

**Comment by lohanidamodar** [comment]
> @lohanidamodar we need to make sure the PR is using https://github.com/utopia-php/image v0.2.1

@eldadfux done

**Comment by eldadfux** [comment]
@christyjacob4 if you're OK we can merge.

**Review by christyjacob4 — APPROVED** [review]
Looks Good @lohanidamodar

**Pull request merged**